tools/cldr-code/src/main/java/org/unicode/cldr/util/SimpleFactory.java

```diff
@@ -775,33 +779,55 @@ public File[] getSourceDirectories() {
         return sourceDirectories;
     }
 
+    LoadingCache<String, List<File>> sourceDirCache =
```

Comment:
How about making this private?

Also, if there are cases where a SimpleFactory is created but getSourceDirectoriesForLocale doesn't get called on it, then maybe lazy initialization would save some time/memory?

Reply:
+1 to private

> where a SimpleFactory is created but getSourceDirectories doesn't get called

THat would be a factory that never has `make()` called on it, which doesn't seem like a common use case, because factories are singletons.